docs/reference/language/modules.md

```diff
@@ -1,189 +1,139 @@
-project folder containing a metadata file named `juvix.yaml`. Each Juvix
-file has to define a <u>module</u> of the same name. The name of the
-module must coincide with the path of its file relative to its
-project's root directory. For example, if the file is
-`root/Data/List.juvix` then the module must be called `Data.List`,
-assuming `root` is the project's folder.
-
-To check that Juvix is correctly detecting your project's root, one can
-run the command `juvix dev root File.juvix`.
+## Top modules
 
-## Importing modules
+A module has a name and a body, which comprises a sequence of
+[statements](statement.md).
 
-To bring into the current scope all module definitions from other
-external modules, one can use the `import` keyword along with the
-corresponding module name. This way, one gets all the imported names
-qualified.
+In order to define a module named `Data.List` we will use the following syntax:
 
 ```juvix
--- A.juvix
-module A;
-   axiom
-   Nat : Type;
+module Data.List;
 
-   axiom
-   zero : Nat;
-end;
-
--- B.juvix
-module B;
-    import A;
-    x : A.Nat;
-    x := A.zero;
+<body>
 ```
 
-Additionally, one can <u>open</u> an imported module making available
-all its names by their unqualified name.
+### Top module naming convention
 
-```juvix
--- A.juvix
-module A;
-   axiom
-   Nat : Type;
+Top modules that belong to a [project](project.md) must follow a naming
+convention. That is, if `dir` is the root of a project, then the module in
+`dir/Data/List.juvix` must be named `Data.List`.
 
-   axiom
-   zero : Nat;
-end;
+## _Import_ and _open_ statements
 
--- B.juvix
-module B;
-    import A;
-    open A;
-    x : Nat;
-    x := zero;
-end;
-```
-
-However, opening modules may create name collisions if you already have
-the imported names as definitions in the current module. In this case,
-Juvix will complain with an error, letting you know which symbols are
-ambiguous. For example, in module `B` below, the name `a` is ambiguous.
+In order to access the definitions from another modules we use an
+_import_ statement. To import some module named `Data.List` we will write
 
 ```juvix
--- A.juvix
-module A;
-axiom
-A : Type;
-
-axiom
-a : A;
-end;
-
--- B.juvix
-module B;
+import Data.List;
+```
 
-import A;
-open A;
+Now, we can access the definitions in the imported module using _qualified
+names_. E.g., `Data.List.sort`.
 
-axiom
-a : A;
+It is possible to import modules and give them a more convinent way thus:
 
-x := a;
-end;
+```juvix
+import Data.List as List;
 ```
 
-One alternative here is hiding the name `a` as follows.
+If we want to access the contents of a module without the need to qualify the
+names, we use an _open statement_. The syntax is as follows:
 
 ```juvix
--- B.juvix
-module B;
-import A;
-open A hiding {a};
-
-axiom
-a : A;
+open Data.List;
+```
 
+Now we can simply write `sort`. It is important to remember that when we
+open a module, that module must be in scope.
```

Comment:
Use the term "imported" as we have already introduced this concept, and we haven't introduced the concept of scope.

```suggestion
open a module, that module must already be imported.
```

Reply:
> Use the term "imported" as we have already introduced this concept, and we haven't introduced the concept of scope.

I used `in scope` instead of `imported` because we can also open local modules. We should think of a more clear way to include both

Reply:
You're right. Do you think - "that module must either be imported or defined as a local module" is too verbose?

Reply:
I've added your suggestion as a clarification for "in scope".